lnrpc/routerrpc/router_server.go

```diff
@@ -401,11 +447,276 @@ func (s *Server) EstimateRouteFee(ctx context.Context,
 	return &RouteFeeResponse{
 		RoutingFeeMsat: int64(route.TotalFees()),
 		TimeLockDelay:  int64(route.TotalTimeLock),
+		FailureReason:  lnrpc.PaymentFailureReason_FAILURE_REASON_NONE,
 	}, nil
 }
 
-// SendToRouteV2 sends a payment through a predefined route. The response of this
-// call contains structured error information.
+// probeInvoice estimates fees along a route to a destination that is specified
+// in an invoice. The estimation duration is limited by a timeout. In case that
+// route hints are provided, this method applies a heuristic to identify LSPs
+// which might block probe payments. In that case, fees are manually calculated
+// and added to the probed fee estimation up until the LSP node. If the route
+// hints don't indicate an LSP, they are passed as arguments to the
+// SendPayment_V2 method, which enable it to send probe payments to the payment
+// request destination.
+func (s *Server) probeInvoice(ctx context.Context, paymentRequest string,
```

Comment:
Is `probePaymentRequest` a better name here?

Reply:
yeah or probePaymentTarget?